backend/src/main/java/de/yourchessboook/service/LichessService.java

```diff
@@ -26,29 +33,101 @@ public LichessGamesDto getGames(String username) {
         LichessGamesDto lichessGamesDto = lichessClient.getGames(username);
 
         List<LichessGameDto> lichessGameDtoList = lichessGamesDto.getLichessGameDtos();
-        if(lichessGameDtoList.isEmpty()){
+        if (lichessGameDtoList.isEmpty()) {
             return lichessGamesDto;
         }
 
         saveToDB(lichessGameDtoList);
 
         return lichessGamesDto;
     }
-    
-    private void saveToDB(List<LichessGameDto> lichessGameDtoList){
+
+    private void saveToDB(List<LichessGameDto> lichessGameDtoList) {
         for (LichessGameDto lichessGameDto : lichessGameDtoList) {
             GameEntity gameEntity = new GameEntity();
             gameEntity.setOpening(lichessGameDto.getOpening().getName());
             gameEntity.setMoves(lichessGameDto.getMoves());
+            gameEntity.setWhitePlayer(lichessGameDto.getPlayers().getWhite().getUser().getName());
+            gameEntity.setBlackPlayer(lichessGameDto.getPlayers().getBlack().getUser().getName());
             gameEntity.setWinner(evaluate(lichessGameDto.getWinner(), lichessGameDto.getPlayers().getWhite().getUser().getName(), lichessGameDto.getPlayers().getBlack().getUser().getName()));
             gameRepository.save(gameEntity);
         }
     }
 
-    private String evaluate(String winner, String white, String black){
-        if(Objects.equals(winner, "white")){
+    private String evaluate(String winner, String white, String black) {
+        if (Objects.equals(winner, "white")) {
             return white;
+        } else if (Objects.equals(winner, "black")) {
+            return black;
+        } else {
+            return "draw";
+        }
+    }
+
+    public ResponseEntity<List<Opening>> getMostFrequentWhiteOpenings(String username) {
+        List<GameEntity> gameEntityList = gameRepository.findAllByWhitePlayerIsOrderByOpening(username);
+
+        return sortAndFilterGames(username, gameEntityList);
+    }
+
+    public ResponseEntity<List<Opening>> getMostFrequentBlackOpenings(String username) {
+        List<GameEntity> gameEntityList = gameRepository.findAllByBlackPlayerIsOrderByOpening(username);
+
+        return sortAndFilterGames(username, gameEntityList);
+    }
+
+    private ResponseEntity<List<Opening>> sortAndFilterGames(String username, List<GameEntity> gameEntityList) {
+        List<Opening> unsortedOpeningList = new ArrayList<>();
+
+        for (GameEntity gameEntity : gameEntityList) {
+            String fullOpeningName = gameEntity.getOpening();
+            String[] splitOpeningName = fullOpeningName.split(": ");
+            String openingName = splitOpeningName[0]; //cut down to prefix
+
+            boolean found = false;
+            for (Opening opening : unsortedOpeningList) { //opening existing?
```

Comment:
alles in eine Methode mit return Boolean

Reply:
sehr sinnvoll, gar nicht dran gedacht - gleich umgesetzt!